server/src/utils/auth.utils.ts

```diff
@@ -0,0 +1,59 @@
+import { Types } from 'mongoose';
+import bcrypt from 'bcryptjs';
+import crypto from 'crypto';
+import * as jwt from 'jsonwebtoken';
+import { OAuth2Client, LoginTicket, TokenPayload } from 'google-auth-library';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const hashInput = async function (input: string): Promise<string> {
+  const salt = await bcrypt.genSalt(10);
+  return await bcrypt.hash(input, salt);
+};
+
+const validateInput = function (input: string, hashedInput: string): Promise<boolean> {
+  return bcrypt.compare(input, hashedInput);
+};
+
+const generateRandomToken = (length: number): string => {
+  return crypto.randomBytes(length || 32).toString('hex');
+};
+
+const generateJwtToken = (userId: Types.ObjectId, email: string, accessType: string): string => {
+  try {
+    const maxAge = 24 * 60 * 60 * 7 * 8; // token stays for 2 month (1 sec increment)
```

Comment:
lets have this maxAge in a separate constant file. so we know where to change constants easily

Reply:
okk